app/tests/pages/eligibility.test.tsx

```diff
@@ -1,58 +1,301 @@
-import { fireEvent, render, screen } from '@testing-library/react'
+import type { EligibilityData, SessionData } from '@customTypes/common'
+import { act, fireEvent, render, screen } from '@testing-library/react'
+import { renderHook } from '@testing-library/react-hooks'
 import { axe } from 'jest-axe'
+import cloneDeep from 'lodash/cloneDeep'
+import mockRouter from 'next-router-mock'
+import singletonRouter, { useRouter } from 'next/router'
+import { useState } from 'react'
+import renderer from 'react-test-renderer'
 
 import Eligibility from '@pages/eligibility'
 
+/**
+ * Test setup
+ */
+
+// Mock the nextjs router
+jest.mock('next/router', () => require('next-router-mock'))
+// This is needed for mocking 'next/link':
+jest.mock('next/dist/client/router', () => require('next-router-mock'))
+// Temporary workaround for all next.js 12.2.0+
+// See active issue: https://github.com/scottrippey/next-router-mock/issues/58
+jest.mock('next/dist/shared/lib/router-context', () => {
+  const { createContext } = require('react')
+  const router = require('next-router-mock').default
+  const RouterContext = createContext(router)
+  return { RouterContext }
+})
+
+// Mock the session
+let mockSession: SessionData
+const emptyMockEligibility: EligibilityData = {
+  residential: '',
+  categorical: [],
+  previouslyEnrolled: '',
+  adjunctive: [],
+}
+const emptyMockSession: SessionData = {
+  chooseClinic: {
+    clinic: undefined,
+    zipCode: '',
+  },
+  contact: {
+    firstName: '',
+    lastName: '',
+    phone: '',
+    comments: '',
+  },
+  eligibility: cloneDeep(emptyMockEligibility),
+  income: {
+    householdSize: ''
+  }
+}
+const setMockSession = jest.fn()
+
+/**
+ * Begin tests
+ */
+
 describe('Eligibility', () => {
-  it('should render first question text', () => {
-    render(<Eligibility previousRoute="/review" />)
+  beforeEach(() => {
+    // Set the router before each test.
+    act(() => {
+      mockRouter.setCurrentUrl('/eligibility')
+    })
 
-    const question = screen.getByText(/Do you live or work in Montana?/i)
+    // Reset the mock session before each test.
+    mockSession = cloneDeep(emptyMockSession)
+  })
 
-    expect(question).toBeInTheDocument()
-    expect(question).toMatchSnapshot()
+  it('should match full page snapshot', () => {
+    const tree = renderer
+      .create(<Eligibility session={mockSession} setSession={setMockSession} />)
+      .toJSON()
+    expect(tree).toMatchSnapshot()
   })
 
   it('should pass accessibility scan', async () => {
-    const { container } = render(<Eligibility previousRoute="/review" />)
-    const results = await axe(container)
+    const { container } = render(
+      <Eligibility session={mockSession} setSession={setMockSession} />
+    )
+
+    // Must call axe() like this to satisfy react testing.
+    let results
+    await act(async () => {
+      results = await axe(container)
+    })
```

Comment:
@karinamzalez I think I figured out how to properly call `act()` in order to fix all those test warnings!